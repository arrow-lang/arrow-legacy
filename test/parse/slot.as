let x: bool;
let y: int = 40;
let z = false;
let mutable w = 10.15;
