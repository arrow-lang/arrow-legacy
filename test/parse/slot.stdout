{
  "tag": "Module",
  "span": "test/parse/slot.as:1.1-11.48",
  "name": "slot",
  "block": {
    "tag": "Block",
    "span": "test/parse/slot.as:1.1-11.48",
    "statements": [
      {
        "tag": "ExternSlot",
        "span": "test/parse/slot.as:1.1-26",
        "exported": true,
        "mutable": false,
        "name": "a",
        "type": {
          "tag": "TypePath",
          "span": "test/parse/slot.as:1.22-25",
          "segments": [
            "int"
          ]
        }
      },
      {
        "tag": "ExternSlot",
        "span": "test/parse/slot.as:2.1-19",
        "exported": false,
        "mutable": false,
        "name": "b",
        "type": {
          "tag": "TypePath",
          "span": "test/parse/slot.as:2.15-18",
          "segments": [
            "int"
          ]
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:3.1-24",
        "exported": true,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/slot.as:3.12-13",
          "text": "c",
          "mutable": false
        },
        "type": {
          "tag": "TypePath",
          "span": "test/parse/slot.as:3.15-18",
          "segments": [
            "int"
          ]
        },
        "initializer": {
          "tag": "Integer",
          "span": "test/parse/slot.as:3.21-23",
          "text": "10"
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:4.1-13",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/slot.as:4.5-6",
          "text": "x",
          "mutable": false
        },
        "type": {
          "tag": "TypePath",
          "span": "test/parse/slot.as:4.8-12",
          "segments": [
            "bool"
          ]
        },
        "initializer": null
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:5.1-17",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/slot.as:5.5-6",
          "text": "y",
          "mutable": false
        },
        "type": {
          "tag": "TypePath",
          "span": "test/parse/slot.as:5.8-11",
          "segments": [
            "int"
          ]
        },
        "initializer": {
          "tag": "Integer",
          "span": "test/parse/slot.as:5.14-16",
          "text": "40"
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:6.1-15",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/slot.as:6.5-6",
          "text": "z",
          "mutable": false
        },
        "type": null,
        "initializer": {
          "tag": "Boolean",
          "span": "test/parse/slot.as:6.9-14",
          "value": false
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:7.1-23",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/slot.as:7.13-14",
          "text": "w",
          "mutable": true
        },
        "type": null,
        "initializer": {
          "tag": "Float",
          "span": "test/parse/slot.as:7.17-22",
          "text": "10.15"
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:8.1-22",
        "exported": false,
        "pattern": {
          "tag": "PatternTuple",
          "span": "test/parse/slot.as:8.5-11",
          "elements": [
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:8.6-7",
              "text": "x",
              "mutable": false
            },
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:8.9-10",
              "text": "y",
              "mutable": false
            }
          ]
        },
        "type": null,
        "initializer": {
          "tag": "Tuple",
          "span": "test/parse/slot.as:8.14-21",
          "elements": [
            {
              "tag": "Integer",
              "span": "test/parse/slot.as:8.15-16",
              "text": "1"
            },
            {
              "tag": "Integer",
              "span": "test/parse/slot.as:8.18-20",
              "text": "20"
            }
          ]
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:9.1-17",
        "exported": false,
        "pattern": {
          "tag": "PatternTuple",
          "span": "test/parse/slot.as:9.5-9",
          "elements": [
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:9.6-7",
              "text": "x",
              "mutable": false
            }
          ]
        },
        "type": null,
        "initializer": {
          "tag": "Tuple",
          "span": "test/parse/slot.as:9.12-16",
          "elements": [
            {
              "tag": "Integer",
              "span": "test/parse/slot.as:9.13-14",
              "text": "4"
            }
          ]
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:10.1-38",
        "exported": false,
        "pattern": {
          "tag": "PatternTuple",
          "span": "test/parse/slot.as:10.5-17",
          "elements": [
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:10.14-15",
              "text": "y",
              "mutable": true
            }
          ]
        },
        "type": {
          "tag": "TypeTuple",
          "span": "test/parse/slot.as:10.19-26",
          "elements": [
            {
              "tag": "TypePath",
              "span": "test/parse/slot.as:10.20-24",
              "segments": [
                "bool"
              ]
            }
          ]
        },
        "initializer": {
          "tag": "Tuple",
          "span": "test/parse/slot.as:10.29-37",
          "elements": [
            {
              "tag": "Boolean",
              "span": "test/parse/slot.as:10.30-35",
              "value": false
            }
          ]
        }
      },
      {
        "tag": "Slot",
        "span": "test/parse/slot.as:11.1-48",
        "exported": false,
        "pattern": {
          "tag": "PatternTuple",
          "span": "test/parse/slot.as:11.5-19",
          "elements": [
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:11.14-15",
              "text": "y",
              "mutable": true
            },
            {
              "tag": "PatternIdentifier",
              "span": "test/parse/slot.as:11.17-18",
              "text": "z",
              "mutable": false
            }
          ]
        },
        "type": {
          "tag": "TypeTuple",
          "span": "test/parse/slot.as:11.21-32",
          "elements": [
            {
              "tag": "TypePath",
              "span": "test/parse/slot.as:11.22-26",
              "segments": [
                "bool"
              ]
            },
            {
              "tag": "TypePath",
              "span": "test/parse/slot.as:11.28-31",
              "segments": [
                "int"
              ]
            }
          ]
        },
        "initializer": {
          "tag": "Tuple",
          "span": "test/parse/slot.as:11.35-47",
          "elements": [
            {
              "tag": "Boolean",
              "span": "test/parse/slot.as:11.36-41",
              "value": false
            },
            {
              "tag": "Integer",
              "span": "test/parse/slot.as:11.43-46",
              "text": "389"
            }
          ]
        }
      }
    ]
  }
}
